lib/questionnaires/model/src/questionnaire_response_model.dart

```diff
@@ -887,4 +895,16 @@ class QuestionnaireResponseModel {
   /// see: [answeredChangeNotifier]
   final invalidityNotifier =
       ValueNotifier<Map<String, String>?>(<String, String>{});
+
+  void handleChangedAnswer(AnswerModel answerModel) {
+    /// By default, [ValueNotifier] will not emit an event if the object not changed
+    /// By calling notifyListeners, we will emit an event by force
+    /// even if the object stored in [answerChangeNotifier] not changed
+    if (answerChangedNotifier.value == answerModel) {
+      answerChangedNotifier.value = answerModel;
+      answerChangedNotifier.notifyListeners();
```

Comment:
> I see that even if the answer has not changed we call notifyListeners. Just for clarification, why is it that we need to fire the event notifiers even if the answer did not change in handleChangedAnswer?

`answerChangedNotifier` stores `AnswerModel` object, so it will not notify the listeners even if the user changes the real answer (`AnswerModel.value`) because the stored object is the same as the incoming object.

> If the value has not changed, do we need to assign it anyway?

You are right, we don't need to assign it.